frontend/src/services/PaymentService.ts

```diff
@@ -2,17 +2,23 @@ import {IPaymentController} from "../models/ControllerTypes";
 import axios from "axios";
 import {ITokenConfig} from "../models/Connection";
 
-export default function PaymentService(config: ITokenConfig): IPaymentController {
+export default function PaymentService(config: ITokenConfig | undefined): IPaymentController {
     return {
         getPayments: (categoryID) => {
-            return axios.get(`/api/payment/?categoryID=${categoryID}`, config).then(response => response.data)
+            return axios.get(`/api/payment/?categoryID=${categoryID}`, config).then(response => {
+                console.log(response.data)
+                return response.data
+            })
         },
         addPayment: payment => {
             return axios.put("/api/payment", payment, config).then(response => response.data)
         },
         deletePayment: (categoryID, paymentID) => {
-            config["data"] = {categoryID, paymentID}
+            config!["data"] = {categoryID, paymentID}
             return axios.delete(`/api/payment`, config).then(response => response.data)
```

Comment:
I would send the ids as pathVariables in the url, Delete normally doesnt have a body